docs/core-concepts/app-deployment.md

```diff
@@ -1,17 +1,255 @@
+
+- It is possible for the .NET Core runtime and libraries to change without your knowledge. In rare cases, this may change the behavior of your app.
+
+## Deploying a Simple Portable App ##
+
+Deploying a portable app with no third-party dependencies simply involves building, testing, and publishing the app. 
+A simple example written in C# illustrates the process. The example uses the dotnet utility from the command line; however, 
+you can also use a development environment, such as Visual Studio or Visual Studio Code, to compile, test, and publish the example.
+
+1. Create a directory for your project, and from the command line, type `dotnet new` to create a new C# console project. 
+
+2. Open the `Program.cs` file in an editor, and replace the auto-generated code with the following code. It prompts the user to enter text, and then displays the individual words entered by the user. It uses the regular expression `\w+` to separate the words in the input text.
+
+    ```cs
+    using System;
+    using System.Text.RegularExpressions;
+
+    namespace Applications.ConsoleApps
+    {
+        public class ConsoleParser
+        {
+            public static void Main()
+            {
+                 Console.WriteLine("Enter any text, followed by <Enter>:\n");
+                 String s = Console.ReadLine();
+                 ShowWords(s);
+                 Console.Write("\nPress any key to continue... ");
+                 Console.ReadKey();
+          }
+
+          private static void ShowWords(String s)
+          {
+              String pattern = @"\w+";
+              var matches = Regex.Matches(s, pattern);
+              if (matches.Count == 0)
+                  Console.WriteLine("\nNo words were identified in your input.");
+              else {
+                  Console.WriteLine("\nThere are {0} words in your string:", matches.Count);
+                  for (int ctr = 0; ctr < matches.Count; ctr++)
+                      Console.WriteLine("   #{0,2}: '{1}' at position {2}", ctr,
+                                        matches[ctr].Value, matches[ctr].Index);
+              }
+          } 
+      }
+    }
+    ```
+
+3. Run the `dotnet restore` command to restore the dependencies specified in your project.
+
+4. Create a debug build of your app by using the `dotnet build` command.
+
+5. After you've debugged and tested the program, you can create the files to be deployed with your app by using the `dotnet publish -c release` command. This creates a release (rather than a debug) version of your app.
+
+   The resulting files are placed in a directory named `publish` that is in a subdirectory of your project's
+`.\bin\release\netcoreapp1.0` subdirectory.
+
+6. Along with your application's files, the publishing process emits a program database (.pdb) file that contains debugging information about your app. You should not package it with your application's files.

```

Comment:
Doesn't this depend on the situation? Deploying PDBs makes debugging exceptions easier, which I think can be useful.